### ISSUE TITLE
Issue in reproducing COCO pretraining results

### ISSUE THREAD

**Issue by YunweiLiDavid** [gpt4vision/OvSGTR#32]
Dear Author:
First of all, thank you very much for disclosing the code of your outstanding work.  However, I have encountered some difficulties in repeating the coco pretraining process.
I followed the `data.md` instructions and used the `tools/language_sgg_grounding.py` script to generate `coco_train2017_triple.json` and `coco_val2017_triple.json`, then ran the coco pretraining. However, my results are far from the performance of the official checkpoint, `vg-pretrain-coco-swint.pth` on HuggingFace.
In the process, I found that the provided `language_sgg_grounding.py` script cannot be directly applied to process the shipped `captions_train2017_triple.json` without errors, so I made some debugging modifications to make it run. Because of that, I'm not sure whether my changes introduced unintended deviations from the original pipeline.
Therefore, it would be extremely helpful if you could open-source the `coco_train2017_triple.json` and `coco_val2017_triple.json` files used for your training (for example via HuggingFace). 
If full release is not feasible, I would greatly appreciate if you could share some details, e.g. a few sample entries (5–10) from each JSON, with all fields included; or key statistical distributions (e.g. number of non-empty relations, triples per image, predicate frequency, bbox / mask availability).
Thank you very much for your time and help!

**Comment by JosephChenHub** [comment]
> Dear Author: First of all, thank you very much for disclosing the code of your outstanding work. However, I have encountered some difficulties in repeating the coco pretraining process. I followed the `data.md` instructions and used the `tools/language_sgg_grounding.py` script to generate `coco_train2017_triple.json` and `coco_val2017_triple.json`, then ran the coco pretraining. However, my results are far from the performance of the official checkpoint, `vg-pretrain-coco-swint.pth` on HuggingFace. In the process, I found that the provided `language_sgg_grounding.py` script cannot be directly applied to process the shipped `captions_train2017_triple.json` without errors, so I made some debugging modifications to make it run. Because of that, I'm not sure whether my changes introduced unintended deviations from the original pipeline. Therefore, it would be extremely helpful if you could open-source the `coco_train2017_triple.json` and `coco_val2017_triple.json` files used for your training (for example via HuggingFace). If full release is not feasible, I would greatly appreciate if you could share some details, e.g. a few sample entries (5–10) from each JSON, with all fields included; or key statistical distributions (e.g. number of non-empty relations, triples per image, predicate frequency, bbox / mask availability). Thank you very much for your time and help!

I was unable to access the original compute  clusters and the original files was erased. If you want to do pre-training using caption data, the first step is to parse triplets from the caption, then do grounding, and post processing like clearing empty triplets should be done before training.

**Comment by MJMJ12138** [comment]
Hello, author. I would like to ask a question. According to what was said above, before pre - training with COCO, I need to use the `tools/language_sgg_grounding.py` file to generate `coco_train2017_triple.json` and `coco_val2017_triple.json`. However, I didn't see any relevant instructions in `data.md`. May I ask how I can find the relevant steps?

**Comment by JosephChenHub** [comment]
> Hello, author. I would like to ask a question. According to what was said above, before pre - training with COCO, I need to use the `tools/language_sgg_grounding.py` file to generate `coco_train2017_triple.json` and `coco_val2017_triple.json`. However, I didn't see any relevant instructions in `data.md`. May I ask how I can find the relevant steps?

I did not upload all relevant files. You are encouraged to do it. `tools/language_sgg_grounding.py`  is enough for this task.
